File: backend/src/main/kotlin/com/alpha/domain/repository/ServiceRepository.kt
package com.alpha.domain.repository

import com.alpha.domain.entity.ServiceEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface ServiceRepository : JpaRepository<ServiceEntity, UUID> {
    @Query("SELECT s FROM ServiceEntity s WHERE s.business.id = :businessId")
    fun findByBusinessId(@Param("businessId") businessId: UUID): List<ServiceEntity>
    
    @Query("SELECT s FROM ServiceEntity s WHERE s.business.id = :businessId AND s.isActive = true")
    fun findByBusinessIdAndIsActiveTrue(@Param("businessId") businessId: UUID): List<ServiceEntity>
    
    @Query("SELECT s FROM ServiceEntity s WHERE s.business.id = :businessId ORDER BY s.sortOrder")
    fun findByBusinessIdOrdered(@Param("businessId") businessId: UUID): List<ServiceEntity>
}

File: backend-scala-zio/src/main/scala/com/alpha/repository/ServiceRepository.scala
package com.alpha.repository

import zio.*
import com.alpha.domain.model.*
import java.util.UUID
import io.getquill.*

trait ServiceRepository:
  def findById(id: UUID): Task[Option[Service]]
  def findByBusinessId(businessId: UUID): Task[List[Service]]
  def findByBusinessIdAndActive(businessId: UUID): Task[List[Service]]
  def create(service: Service): Task[UUID]
  def update(service: Service): Task[Int]
  def delete(id: UUID): Task[Int]

object ServiceRepository:
  val layer: ZLayer[PostgresCtx, Nothing, ServiceRepository] = 
    ZLayer.fromFunction(new ServiceRepositoryImpl(_))

class ServiceRepositoryImpl(ctx: PostgresCtx) extends ServiceRepository:

  import ctx.*

  override def findById(id: UUID): Task[Option[Service]] = 
    ZIO.attempt:
      run(query[Service].filter(_.id == lift(id))).headOption

  override def findByBusinessId(businessId: UUID): Task[List[Service]] = 
    ZIO.attempt:
      run(query[Service].filter(_.businessId == lift(businessId))).toList

  override def findByBusinessIdAndActive(businessId: UUID): Task[List[Service]] = 
    ZIO.attempt:
      run(query[Service].filter(s => s.businessId == lift(businessId) && s.isActive == lift(true))).toList

  override def create(service: Service): Task[UUID] = 
    ZIO.attempt:
      run(query[Service].insertValue(lift(service)))
      service.id

  override def update(service: Service): Task[Int] = 
    ZIO.attempt:
      run(query[Service]
        .filter(_.id == lift(service.id))
        .updateValue(lift(service)))
      1

  override def delete(id: UUID): Task[Int] = 
    ZIO.attempt:
      run(query[Service].filter(_.id == lift(id)).delete)
      1
